src/Symfony/Component/Serializer/Mapping/Loader/AttributeLoader.php

```diff
@@ -98,23 +98,26 @@ public function loadClassMetadata(ClassMetadataInterface $classMetadata): bool
                 }
 
                 foreach ($this->loadAttributes($property) as $annotation) {
+                    $loaded = true;
+
+                    $attributeMetadata = $attributesMetadata[$property->name];
+
                     if ($annotation instanceof Groups) {
                         foreach ($annotation->getGroups() as $group) {
-                            $attributesMetadata[$property->name]->addGroup($group);
+                            $attributeMetadata->addGroup($group);
                         }
-                    } elseif ($annotation instanceof MaxDepth) {
-                        $attributesMetadata[$property->name]->setMaxDepth($annotation->getMaxDepth());
-                    } elseif ($annotation instanceof SerializedName) {
-                        $attributesMetadata[$property->name]->setSerializedName($annotation->getSerializedName());
-                    } elseif ($annotation instanceof SerializedPath) {
-                        $attributesMetadata[$property->name]->setSerializedPath($annotation->getSerializedPath());
-                    } elseif ($annotation instanceof Ignore) {
-                        $attributesMetadata[$property->name]->setIgnore(true);
-                    } elseif ($annotation instanceof Context) {
-                        $this->setAttributeContextsForGroups($annotation, $attributesMetadata[$property->name]);
+
+                        continue;
                     }
 
-                    $loaded = true;
+
+                    match (true) {
```

Comment:
this is broken. When using the annotation reader, the `loadAttributes` method will include *all* annotations of the reflector (unlike attributes which are filtered for known attributes). The existing code is silently ignoring those (there is an implicit no-op `else` case). Your replacement makes it an Error instead (as `match` is required to match a branch), which makes this loader incompatible with classes that use *other* annotations when annotation support is enabled in the serializer component.